test/integration/controller_test.go

```diff
@@ -417,6 +417,148 @@ func getLastOperationResponseByPollCountStates(numOfResponses int, stateProgress
 	return getLastOperationResponseByPollCountReactions(numOfResponses, reactionProgressions)
 }
 
+// newTestController creates a new test controller injected with fake clients
+// and returns:
+//
+// - a fake kubernetes core api client
+// - a fake service catalog api client
+// - a fake osb client, with configuration for happy path testing
+// - a test controller
+// - the shared informers for the service catalog v1beta1 api
+// - a function for shutting down the API server
+// - a function for shutting down the controller.
+//
+// If there is an error, newTestController calls 'Fatal' on the injected
+// testing.T.
+func newControllerTestTestController(ct *controllerTest) (
```

Comment:
It's not clear to me what the difference is between this and `newTestController` - and we should find a less ambiguous name for this func.

Reply:
This moves the setup for the 'ControllerTest' struct inside the underlying TestController setup to avoid races caused by setting up the controllers after starting them. I didn't want to rewrite the whole thing at this time until we were deciding what to do with the overlying cluster id issue.

Reply:
I finally get the name, but it broke my brain. However, after the PR languishing for so long, I think you get a pass. Sorry for the wait.